nifi-nar-bundles/nifi-elasticsearch-bundle/nifi-elasticsearch-processors/src/main/java/org/apache/nifi/processors/elasticsearch/AbstractElasticsearchProcessor.java

```diff
@@ -74,8 +83,9 @@
         Set<ValidationResult> results = new HashSet<>();
 
         // Ensure that if username or password is set, then the other is too
-        Map<PropertyDescriptor, String> propertyMap = validationContext.getProperties();
-        if (StringUtils.isEmpty(propertyMap.get(USERNAME)) != StringUtils.isEmpty(propertyMap.get(PASSWORD))) {
+        String userName = validationContext.getProperty(USERNAME).evaluateAttributeExpressions().getValue();
+        String password = validationContext.getProperty(PASSWORD).evaluateAttributeExpressions().getValue();
```

Comment:
@mattyb149 why do you say that "Sensitive properties don't support EL yet? There are multiple instances of them using it, see below. My understanding is just that the sensitive property will just change how it's stored and displayed back to the user. It doesn't stop the value from getting interpreted as EL. 

It would be true that it may not make sense yet from the registry perspective (getting stored in plaintext) but there are other reasons to evaluate EL on a sensitive property.

https://github.com/apache/nifi/blob/6dc2f14198bb475ace8e762e348ca5294400cccb/nifi-nar-bundles/nifi-a...
https://github.com/apache/nifi/blob/3a7ddc6a35f143c1d4a735226fe7467a83d04583/nifi-bootstrap/src/main...
https://github.com/apache/nifi/blob/4fc0e9c40763d05eee01b48f76cd4cf273b61330/nifi-nar-bundles/nifi-s...